### ISSUE TITLE
false positive on ++ and -- in `operator-no-unspaced`

### ISSUE THREAD

**Issue by evilebottnawi** [kristerkari/stylelint-scss#286]
Input:
```css
a {
  width: ++1;
}
```

Require:
```css
a {
  width: + + 1;
}
```

Expected (no errors):
```css
a {
  width: ++1;
}
```

**Comment by kristerkari** [comment]
Thanks! Would you have any other examples for the bug? I don't think that it would be very common to write `++1` as the value as it is turned to just `1` by Sass.

**Comment by clarfonthey** [comment]
Want to also add to this that unary negation is also a false positive here, and `-$margin-offset` also throws the error, which seems more likely to use.

**Comment by fluidsonic** [comment]
I ran straight into this issue with the unary minus when adding `stylelint` to a project :)

<img width="646" alt="image" src="https://github.com/stylelint-scss/stylelint-scss/assets/85242/3f86d82e-437e-4370-87f3-4915a6a1abb2">

**Comment by kristerkari** [comment]
@clarfonthey finally managed to get this fixed.

@fluidsonic I did not fix your example since it seems that either parens or interpolation is needed to make the negation to work when using multiple variables.

https://github.com/stylelint-scss/stylelint-scss/releases/tag/v6.3.2

**Comment by fluidsonic** [comment]
Thanks, I noticed it behaves very differently in SASS than in other languages, and from what I'd expect. So the problem is with SASS and the warning here just highlights that it's an odd case, which is good 👍

**Comment by clarfonthey** [comment]
Hmm, are you sure this is fixed? I'm still getting this error; here's a minimal repro:

```scss
$lhs: 1;
$rhs: 2;
@if $lhs <= -$rhs {}
```

**Comment by clarfonthey** [comment]
Hmm, are you sure this is fixed? I'm still getting this error; here's a minimal repro:

```scss
$lhs: 1;
$rhs: 2;
@if $lhs <= -$rhs {}
```

@kristerkari

**Comment by kristerkari** [comment]
@clarfonthey The `operator-no-unspaced` rule uses a custom made parser to figure out if the operators should have spacing or not. This means that there are a lot of false posivites, and your example is probably one of them.

I'll try to fix the example that you provided 👍 

For anyone else that wants to report more false positives, please open a new Github issue with an example 🙇
